### PR TITLE
Golangci-lint integration and errors fixes

### PR DESCRIPTION
**Pull request by AugustasV**
Summary
#55 Integrate GoLangici-lint and fix most of linter errors.

#### Ticket Link

  Integrate GolangCI lint https://github.com/mattermost/mattermost-plugin-skype4business/issues/55

### CODE REVIEW

**Review by hanzei — APPROVED** [review]
Nice, thanks for migrating also this plugin :+1:

**Comment by jasonblais** [comment]
@mickmister @kosgrz kind reminder to help with a review :)

**Review by mickmister — APPROVED** [review]
LGTM, thanks @AugustasV! I have one request on changing spelling of `custome` -> `custom`

There are some merge conflicts from my PR that sync'd this repo with the starter template. Let me know if you have any trouble resolving the conflicts

**Comment by AugustasV** [comment]
@hanzei @mickmister done, conflicts solved.

File: Makefile [hunk 1]
```diff
@@ -93,6 +93,14 @@ else
...
 endif
 
+## Builds the webapp in debug mode, if it exists.
+.PHONY: webapp-debug
+webapp-debug: webapp/.npminstall
+ifneq ($(HAS_WEBAPP),)
+	cd webapp && \
+	$(NPM) run debug;
+endif
+
```
Comment (hanzei) [inline Makefile:103]: That seems like an unrelated change.

File: go.mod [hunk 2]
```diff
@@ -3,7 +3,17 @@ module github.com/mattermost/mattermost-plugin-skype4business
 go 1.13
 
 require (
+	github.com/google/go-cmp v0.5.2 // indirect
```
Comment (hanzei) [inline go.mod:6]: Could you please remove the `indirect` dependencies?

**Review by hanzei — CHANGES_REQUESTED** [review]
Two things got mixed in the Merge

**Comment by jfrerich** [comment]
Hi @AugustasV, how are things going with this PR?  Do you have any questions or need any help addressing the feedback from @hanzei?

**Review by hanzei — APPROVED** [review]
Build tests locally and does deploy correctly. LGMT :+1:

**Comment by hanzei** [comment]
Skipping QA review as these are only build changes

**Pull request merged**